Demo/Demo/FeatureViewControllers/CardDemoViewController.swift

```diff
@@ -88,6 +90,25 @@ class CardDemoViewController: FeatureBaseViewController {
 
     @objc func didTapPayButton() {
         updateTitle("Processing order...")
-        // TODO: Call `processOrder`
+        payButton.startAnimating()
+        // TODO: Call `processOrder` and `payButton.stopAnimating()` once process order is complete
+    }
+
+    @objc private func editingChanged() {
+        cardNumberTextField.text = cardFormatter.formatCardNumber(cardNumberTextField.text ?? "")
+        expirationTextField.text = cardFormatter.formatExpirationDate(expirationTextField.text ?? "")
+
+        guard
+            let cardNumberCount = cardNumberTextField.text?.count,
+            let expirationDateCount = expirationTextField.text?.count,
+            let cvvCount = cvvTextField.text?.count,
+            cardNumberCount >= 18 && cardNumberCount <= 19,
```

Comment:
should we do `18` here, should it be `15`? 
I'm not sure what the required length for each card type is, but I do see you have `americanExpress` or `visa` which can be 15 or 16 :smile:

Reply:
we can also make sure `cardNumberCount` here is `>= 4`, then we can use `CardType` to `getCardType()` then we can assert the exact length / or length range of the specific card type.

But this is just a suggestion, not a blocker, I'm fine either way :smile: